### PR TITLE
Improve serialisation

### PR DESCRIPTION
**Pull request by EdsonAlcala**
**What changed? Why?**

This PR includes many improvements with regards to serialisation and usage of the library. Including full support for Near actions.

The changes include:
 
- update near-sdk to 5.3.0
- included integration test for Near transactions
- improve serialisation of near structs so rust clients, json clients and near contracts can consume the library
- added support for:
  - CreateAccountAction
  - DeployContractAction
  - FunctionCallAction
  - TransferAction
  - StakeAction
  - AddKeyAction
  - DeleteKeyAction
  - DeleteAccountAction
- update readme with link to examples
- added `from_json` function to enable raw json strings to be converted into a NearTransaction
- added `build_with_signature` function to enable building a signed payload
- included utilities for public keys
- included utilities for signatures
- split types module into multiple files with their correspondent tests:
  - actions
  - integers
  - block_hash
  - signature
  - public_key
- increased test coverage in general, added more test cases

**Notes to reviewers**

The integration test for Near is using 
```rust
near-jsonrpc-client = { git = "https://github.com/edsonalcala/near-jsonrpc-client-rs.git", branch = "master" }
```
Since it was required a "send_raw_tx", instead of a typed SignedTransaction, I forked the `near-json-rpc-client` to make it work (although I'll try to get that merged into the `near-json-rpc-client`) I thought it was mentioning it.

**How has it been tested?**

The library itself includes unit tests and 2 integration tests. However, the library was tested using the examples repository https://github.com/Omni-rs/examples, that includes a set of NEAR smart contracts with an integration test running against near-workspaces.

So far all the tests passed:

Screenshot 2024-09-06 at 14 35 56

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,13 +33,18 @@ alloy-rlp = { version = "0.3.8", features = ["derive"] }
...
 # near
-near-primitives = { version = "0.24.1" }
-near-crypto = { version = "0.24.1" }
+near-primitives = { version = "0.25.0" }
+near-crypto = { version = "0.25.0" }
+near-workspaces = { version = "0.12.0", features = [
+    "experimental",
+    "unstable",
+] }
+near-jsonrpc-client = { git = "https://github.com/edsonalcala/near-jsonrpc-client-rs.git", branch = "master" }
```
Comment (ilblackdragon) [inline Cargo.toml:42]: let's get a fixed/tag version? otherwise this going to break for someone

File: src/near/types/actions.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
+use crate::near::types::PublicKey;
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::AccountId;
+
+use super::{U128, U64};
+
+#[derive(Serialize, Deserialize, Debug, Clone, BorshSerialize, BorshDeserialize, PartialEq, Eq)]
+#[serde(crate = "near_sdk::serde")]
+pub enum Action {
```
Comment (ilblackdragon) [inline src/near/types/actions.rs:10]: is there no library from nearcore to reuse here? or it's too many dependencies there?

**Review by ilblackdragon — APPROVED** [review]
Looks good overall.

File: Cargo.toml (hunk 1 above)
Comment (EdsonAlcala) [reply Cargo.toml]: yes, fixed

**Pull request merged**